lib/faraday/adapter.rb

```diff
@@ -34,9 +34,10 @@ def call(env)
       env.clear_body if env.needs_body?
     end
 
-    def save_response(env, status, body, headers = nil)
+    def save_response(env, status, body, headers = nil, reason_phrase = "")
       env.status = status
       env.body = body
+      env.reason_phrase = reason_phrase.to_s.strip
```

Comment:
Could the reason phrase be nil if it's missing, instead of an empty string? That way users can test the presence of the phrase more easily.